Conflict between Quarkus bom and temporal grpc version

Quarkus bom contains a different grpc version than temporal.

This is making the mock environment fail with the following error:

```
Caused by: java.lang.NoClassDefFoundError: io/grpc/inprocess/InProcessServerBuilder
	at io.temporal.testserver.TestServer.createServer(TestServer.java:119)
	at io.temporal.testing.TestWorkflowEnvironmentInternal.<init>(TestWorkflowEnvironmentInternal.java:91)
	at io.temporal.testing.TestWorkflowEnvironment.newInstance(TestWorkflowEnvironment.java:100)
	at fr.lavachequicode.temporal.test.plugin.TestWorkflowRecorder.createTestWorkflowEnvironment(TestWorkflowRecorder.java:12)
	at io.quarkus.deployment.steps.TemporalTestProcessor$recordTestEnvironment521213906.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.TemporalTestProcessor$recordTestEnvironment521213906.deploy(Unknown Source)
```

This project currently include a temporary workaround that allow to pass the build (but probably won't work for actual users since this would be overridden again by the bom present in their project):

https://github.com/quarkiverse/quarkus-temporal/blob/56b66d4f0b1dddbda0044392d8fffeb49ad849c7/pom.xml#L56-L90


the goal of this issue is to find a more permanent solution to this issue

Ideas: 
- create a shaded version of temporal containing the right grpc version
- modify quarkus bom not to include grpc version
- work with temporal to update to the latest grpc version
- @melloware do you have any other ideas here ?

--- a/extension/deployment/pom.xml
+++ b/extension/deployment/pom.xml
@@ -14,6 +14,10 @@
             <artifactId>quarkus-arc-deployment</artifactId>
         </dependency>
         <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-grpc-common-deployment</artifactId>
+        </dependency>
+        <dependency>
             <groupId>io.quarkiverse.temporal</groupId>
             <artifactId>quarkus-temporal</artifactId>
             <version>${project.version}</version>

--- a/extension/runtime/pom.xml
+++ b/extension/runtime/pom.xml
@@ -14,8 +14,24 @@
             <artifactId>quarkus-arc</artifactId>
         </dependency>
         <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-grpc-common</artifactId>
+        </dependency>
+        <dependency>
             <groupId>io.temporal</groupId>
             <artifactId>temporal-sdk</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.grpc</groupId>
+            <artifactId>grpc-inprocess</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.grpc</groupId>
+            <artifactId>grpc-services</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.grpc</groupId>
+            <artifactId>grpc-netty-shaded</artifactId>
         </dependency>
     </dependencies>
     <build>

--- a/integration-tests/pom.xml
+++ b/integration-tests/pom.xml
@@ -33,6 +33,23 @@
             <groupId>io.quarkiverse.temporal</groupId>
             <artifactId>quarkus-temporal-test</artifactId>
             <version>${project.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkiverse.temporal</groupId>
+            <artifactId>quarkus-temporal-test-deployment</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>io.quarkiverse.temporal</groupId>
+            <artifactId>quarkus-temporal-deployment</artifactId>
+            <version>${project.version}</version>
+            <exclusions>
+                <exclusion>
+                    <groupId>*</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
         </dependency>
         <dependency>
             <groupId>io.quarkus</groupId>

--- a/pom.xml
+++ b/pom.xml
@@ -31,7 +31,6 @@
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <quarkus.version>3.13.0</quarkus.version>
         <temporal.version>1.24.3</temporal.version>
-        <grpc.version>1.54.1</grpc.version>
     </properties>
     <dependencyManagement>
         <dependencies>
@@ -46,47 +45,17 @@
                 <groupId>io.temporal</groupId>
                 <artifactId>temporal-sdk</artifactId>
                 <version>${temporal.version}</version>
+                <exclusions>
+                    <exclusion>
+                        <groupId>io.grpc</groupId>
+                        <artifactId>*</artifactId>
+                    </exclusion>
+                </exclusions>
             </dependency>
             <dependency>
                 <groupId>io.temporal</groupId>
                 <artifactId>temporal-testing</artifactId>
                 <version>${temporal.version}</version>
-            </dependency>
-
-            <dependency>
-                <groupId>io.grpc</groupId>
-                <artifactId>grpc-api</artifactId>
-                <version>${grpc.version}</version>
-            </dependency>
-            <dependency>
-                <groupId>io.grpc</groupId>
-                <artifactId>grpc-stub</artifactId>
-                <version>${grpc.version}</version>
-            </dependency>
-            <dependency>
-                <groupId>io.grpc</groupId>
-                <artifactId>grpc-netty-shaded</artifactId>
-                <version>${grpc.version}</version>
-            </dependency>
-            <dependency>
-                <groupId>io.grpc</groupId>
-                <artifactId>grpc-context</artifactId>
-                <version>${grpc.version}</version>
-            </dependency>
-            <dependency>
-                <groupId>io.grpc</groupId>
-                <artifactId>grpc-services</artifactId>
-                <version>${grpc.version}</version>
-            </dependency>
-            <dependency>
-                <groupId>io.grpc</groupId>
-                <artifactId>grpc-protobuf-lite</artifactId>
-                <version>${grpc.version}</version>
-            </dependency>
-            <dependency>
-                <groupId>io.grpc</groupId>
-                <artifactId>grpc-core</artifactId>
-                <version>${grpc.version}</version>
             </dependency>
         </dependencies>
     </dependencyManagement>

